libafl/src/corpus/mod.rs

```diff
@@ -66,26 +66,38 @@ impl From<CorpusId> for usize {
 /// Utility macro to call `Corpus::random_id`
 #[macro_export]
 macro_rules! random_corpus_id {
-    ($corpus:expr, $rand:expr) => {{
-        let cnt = $corpus.count() as u64;
+    ($corpus:expr, $rand:expr, $incl_disabled:expr) => {{
+        let cnt: u64;
+        if $incl_disabled == true {
+            cnt = $corpus.count_with_disabled() as u64;
+        } else {
+            cnt = $corpus.count() as u64;
+        }
         let nth = $rand.below(cnt) as usize;
+
         $corpus.nth(nth)
     }};
 }
 
 /// Corpus with all current [`Testcase`]s, or solutions
 pub trait Corpus: UsesInput + Serialize + for<'de> Deserialize<'de> {
-    /// Returns the number of elements
+    /// Returns the number of elements excluding disabled entries
     fn count(&self) -> usize;
 
+    /// Returns the number of elements including disabled entries
+    fn count_with_disabled(&self) -> usize;
```

Comment:
probably an extra `count_disabled` fn would be nice (else you'll have to do your own arithmetics)

Reply:
And then maybe rename this fn to `count_all`